src_c/event.c

```diff
@@ -632,6 +644,24 @@ dict_from_event(SDL_Event *event)
             _pg_insobj(dict, "start", PyLong_FromLong(event->edit.start));
             _pg_insobj(dict, "length", PyLong_FromLong(event->edit.length));
             break;
+        case SDL_CONTROLLERAXISMOTION:
+            /* https://wiki.libsdl.org/SDL_ControllerAxisEvent */
+            _pg_insobj(dict, "joy", PyInt_FromLong(event->caxis.which));
+            _pg_insobj(dict, "axis", PyLong_FromLong(event->caxis.axis));
+            _pg_insobj(dict, "value", PyLong_FromLong(event->caxis.value));
+            break;
+        case SDL_CONTROLLERBUTTONDOWN:
+        case SDL_CONTROLLERBUTTONUP:
+            /* https://wiki.libsdl.org/SDL_ControllerButtonEvent */
+            _pg_insobj(dict, "joy", PyInt_FromLong(event->cbutton.which));
```

Comment:
Should probably use PyLong_FromLong() to guarantee that it may contain at least 32 bits.